app/survey_config/census_config.py

```diff
@@ -57,11 +57,17 @@ class CensusSurveyConfig(
         ],
         compare=False,
     )
-    data_layer: Iterable[Mapping] = field(
-        default_factory=lambda: [{"nisra": False}], compare=False
-    )
+
     survey_title: LazyString = lazy_gettext("Census 2021")
     sign_out_button_text: str = lazy_gettext("Save and complete later")
+    _is_nisra: bool = False
+
+    def get_data_layer(self, _tx_id: Optional[str] = None) -> list[dict]:
+        return (
+            [{"nisra": self._is_nisra}, {"tx_id": _tx_id}]
+            if _tx_id
+            else [{"nisra": self._is_nisra}]
+        )
```

Comment:
- How come `tx_id` is prefxied with `_`?.
- I think having one data var would be better here. 
```py
        data_layer = [{"nisra": self._is_nisra}]
        if _tx_id:
            data_layer.append({"tx_id": _tx_id})

        return data_layer
```

Reply:
Prefixed with `_` to avoid `W0237: Parameter '_tx_id' has been renamed to 'tx_id' in overridden 'CensusSurveyConfig.get_data_layer' method (arguments-renamed)`. Will use your suggestion code, thanks!